packages/firestore/src/remote/remote_store.ts

```diff
@@ -814,10 +818,20 @@ export class RemoteStore {
   handleUserChange(user: User): Promise<void> {
     log.debug(LOG_TAG, 'RemoteStore changing users: uid=', user.uid);
 
-    // Tear down and re-create our network streams. This will ensure we get a fresh auth token
-    // for the new user and re-fill the write pipeline with new mutations from the LocalStore
-    // (since mutations are per-user).
-    this.disableNetworkInternal(OnlineState.Unknown);
-    return this.enableNetwork();
+    if (this.isNetworkEnabled()) {
+      // Tear down and re-create our network streams. This will ensure we get a fresh auth token
+      // for the new user and re-fill the write pipeline with new mutations from the LocalStore
+      // (since mutations are per-user).
+      this.disableNetworkInternal(OnlineState.Unknown);
+      return this.enableNetwork();
+    } else {
+      // The network is disabled, so we don't want to reconnect. Remain in
+      // 'failed' state. This assert *should* only ever trigger if shutdown() is
+      // called without first removing the user change listener.
+      assert(
+        this.watchStreamOnlineState == OnlineState.Failed,
+        'Since the network is disabled, expected to be in failed state'
+      );
```

Comment:
I think this assert should hold, but it feels a bit out-of-place.  I don't think it's necessary for this method to function correctly, so it's kind of a "By the way, this should be true..." assert.  We could consider verifying that methods aren't called after shutdown(), but to do that I think we should add an `assertNotShutdown()` method and call it from every method... though I'm not sure that's necessary (and if we do it, we should do it across all of our shutdown-able components).

Reply:
I could replace it with a now-idempotent call to disableInternal w/ state == failed, or leave it out entirely. I think it's at least worth keeping a comment there, since it might not be immediately obvious why we're only doing something if the network is enabled.

Reply:
FWIW- I would just do:

```
// Ensure we don't start the network if it's currently disabled.
if (this.isNetworkEnabled()) {
   this.disableNetworkInternal(OnlineState.Unknown);
   return this.enableNetwork();
} else {
  return Promise.resolve();
}
```

Reply:
Ok, this is done now.